File: MGCBack/src/main/java/com/mygeekcollection/backend/entity/Item.java
package com.mygeekcollection.backend.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "collection")
public class Item {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    @JsonIgnore
    private User user;

    @ManyToOne
    @JoinColumn(name = "videogame_id")
    private Videogame videogame;

    @Column(name = "general_condition")
    private String generalCondition;

    @Column(name = "box")
    private Boolean box;

    @Column(name = "manual")
    private Boolean manual;

}

File: MGCFront/src/app/services/user/interfaces/item.ts
import {User} from "./user";
import {Videogame} from "../../videogame/interfaces/videogame";

export class Item {
  id:number;
  user:User;
  videogame:Videogame;
  generalCondition:String;
  box:Boolean;
  manual:Boolean;
}
